### PR TITLE
Allows for a selector in the update function

### PR DESCRIPTION
**Pull request by sscaff1**
The use case is trying to update an array in a document.

### CODE REVIEW

**Comment by marshallswain** [comment]
@sscaff1 Thank you.  It looks like JShint isn't happy, so the tests aren't running.  Do you mind fixing this part:

src/service.js: line 148, col 85, Strings must use singlequote.
1 error

You should also be able to run the tests locally before pushing by doing `npm test`.

**Comment by marshallswain** [comment]
Also, it would be really good to add a test to make sure your use case continues to work in the future.  If you feel up to making the test, that would be amazing.  Otherwise, we can probably tackle it when we're ready to merge.

**Comment by sscaff1** [comment]
Sorry, I fixed it.

File: src/service.js [hunk 1]
```diff
@@ -144,12 +144,14 @@ class Service {
       // previous value. This prevents orphaned documents.
       data[this.id] = id;
     }
+    
+    const selector = Object.assign({}, { [this.id]: id }, typeof params === 'object' ? params : undefined);
```
Comment (daffl) [inline src/service.js]: I think using `params.query` here would be better since that is what all other adapters are using.

**Comment by daffl** [comment]
Just one comment and if we could add a simple test for it we'd be good to go. Just something simple like this:

```js
it('allows to update with a selector', function (done) {
  people.create({ name: 'David', age: 300 }).then(person => {
    people.update({ name: 'David' }, { name: 'Updated David' }).then(otherPerson => {
      assert.equal(person._id, otherPerson._id);
      return people.remove(otherPerson._id);
    }).then(done, done);
  });
});
```

File: src/service.js (hunk 1 above)
Comment (daffl) [inline src/service.js]: Actually shouldn't the selector be what `id` is now?

**Comment by ekryski** [comment]
Yes, thanks @sscaff1 for the PR! I think we do need to add a test for this. Especially because I'm not seeing why updating an array inside a document is not already possible. Wth an `update` you should be passing the whole updated document, which can include the new item in the nested array.

We assume ID because `update` maps to the route structure `PUT /people/:id`. There are only 2 reasons I see the need for doing an update by `params.query`:

1. Is if you want to update multiple documents (which the service adapters intentionally don't support). 

2. You need to update by a key other than your primary id (which you can do by setting the `id`field on initializing the service). However, if that `id` is not unique then you might run into weird results.

Maybe I'm just not fully understanding the use case but could you provide a test or example payload for what you wan to accomplish?

**Comment by sscaff1** [comment]
Hey guys - I'm sorry I haven't gotten back to this. Thank you @ekryski, @daffl and @marshallswain for the feedback. I will update the pull request when I get some time (probably next week) and make sure to include a test and address the feedback above. I appreciate your help and I didn't want you guys to think that I was going to leave this out to dry. Thanks again.

**Comment by ekryski** [comment]
I think #113 will help out a bit but we intentionally don't allow updates for multiple documents. That is what patch is used for. Since this is 3 months stale I'm going to close it.

**Pull request closed without merging**